src/main/java/com/minecolonies/colony/buildings/BuildingMiner.java

```diff
@@ -30,83 +32,90 @@
      */
-    private static final String TAG_SLOCATION      = "shaftLocation";
+    private static final String  TAG_SLOCATION      = "shaftLocation";
     /**
      * The NBT Tag to store the vector-x of the shaft
      */
-    private static final String TAG_VECTORX        = "vectorx";
+    private static final String  TAG_VECTORX        = "vectorx";
     /**
      * The NBT Tag to store the vector-z of the shaft
      */
-    private static final String TAG_VECTORZ        = "vectorz";
+    private static final String  TAG_VECTORZ        = "vectorz";
     /**
      * The NBT Tag to store the location of the cobblestone at the shaft.
      */
-    private static final String TAG_CLOCATION      = "cobblelocation";
+    private static final String  TAG_CLOCATION      = "cobblelocation";
     /**
      * The NBT Tag to store the active node the miner is working on.
      */
-    private static final String TAG_ACTIVE         = "activeNodeint";
+    private static final String  TAG_ACTIVE         = "activeNodeint";
     /**
      * The NBT Tag to store the current level the miner is working in.
      */
-    private static final String TAG_CURRENT_LEVEL  = "currentLevel";
+    private static final String  TAG_CURRENT_LEVEL  = "currentLevel";
     /**
      * The NBT Tag to store the starting node.
      */
-    private static final String TAG_SN             = "StartingNode";
+    private static final String  TAG_SN             = "StartingNode";
     /**
      * The NBT Tag to store the location of the ladder.
      */
-    private static final String TAG_LLOCATION      = "ladderlocation";
+    private static final String  TAG_LLOCATION      = "ladderlocation";
     /**
      * The NBT Tag to store if a ladder has been found yet.
      */
-    private static final String TAG_LADDER         = "found_ladder";
+    private static final String  TAG_LADDER         = "found_ladder";
+
+    private static final String TAG_SHAFT_BLOCK     = "shaftBlock";
+
     /**
      * The maximum upgrade of the building.
      */
-    private static final int    MAX_BUILDING_LEVEL = 3;
+    private static final int         MAX_BUILDING_LEVEL = 3;
     /**
      * The job description.
      */
-    private static final String MINER              = "Miner";
+    private static final String      MINER              = "Miner";
     /**
      * True if shaft is at bottom limit
      */
-    public boolean clearedShaft = false;
+    public               boolean     clearedShaft       = false;
+    /**
+     * Defines the material used for the structure of the horizontal shaft.
+     */
+    private              Block       shaftBlock         = Blocks.PLANKS;
     /**
      * Defines the material used for the floor of the shaft.
      */
-    private Block floorBlock = Blocks.planks;
+    private static final IBlockState floorBlock = Blocks.WOODEN_SLAB.getDefaultState().withProperty(Blocks.WOODEN_SLAB.HALF, BlockSlab.EnumBlockHalf.TOP);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this instance-reference to a static reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS2209)